atc/exec/build/repository.go

```diff
@@ -1,71 +1,68 @@
-package artifact
+package build
 
 import (
 	"sync"
 
-	"github.com/concourse/concourse/atc/worker"
+	"github.com/concourse/concourse/atc/runtime"
 )
 
-// Name is just a string, with its own type to make interfaces using it
+// ArtifactName is just a string, with its own type to make interfaces using it
 // more self-documenting.
-type Name string
+type ArtifactName string
 
-// Repository is the mapping from a Name to an ArtifactSource.
+// Repository is the mapping from a ArtifactName to an Artifact.
 // Steps will both populate this map with new artifacts (e.g.  the resource
 // fetched by a Get step), and look up required artifacts (e.g.  the inputs
 // configured for a Task step).
 //
 // There is only one ArtifactRepository for the duration of a build plan's
 // execution.
 //
-// ArtifactRepository is, itself, an ArtifactSource. As an ArtifactSource it acts
-// as the set of all ArtifactSources it contains, as if they were each in
-// subdirectories corresponding to their ArtifactName.
 type Repository struct {
-	repo  map[Name]worker.ArtifactSource
+	repo  map[ArtifactName]runtime.Artifact
 	repoL sync.RWMutex
 }
 
 // NewArtifactRepository constructs a new repository.
 func NewRepository() *Repository {
 	return &Repository{
-		repo: make(map[Name]worker.ArtifactSource),
+		repo: make(map[ArtifactName]runtime.Artifact),
 	}
 }
 
-//go:generate counterfeiter . RegisterableSource
-// A RegisterableSource	artifact is an ArtifactSource which can be added to the registry
-type RegisterableSource interface {
-	worker.ArtifactSource
+//go:generate counterfeiter . RegisterableArtifact
```

Comment:
I'm guessing that we had to name this something other than "Artifact" to avoid token clash, but this implies that there exists a world in which other artifacts exist which cannot be registered. This name (and the historical naming) is more confusing than not. Could we use a different descriptor that reveals more about what this will eventually be used for? Like... "PotentialBuildArtifact", or at least something that links to some Concourse domain concept?